pandas/core/sparse/array.py

```diff
@@ -789,7 +790,16 @@ def make_sparse(arr, kind='block', fill_value=None):
         if is_string_dtype(arr):
             arr = arr.astype(object)
 
-        mask = arr != fill_value
+        if is_object_dtype(arr.dtype):
+            mask = []
+            for e in arr:
+                if type(e) is type(fill_value):
+                    mask.append(e != fill_value)
+                else:
+                    mask.append(True)
+            mask = np.array(mask)
```

Comment:
this is very awkward

you can do this by starting with an array of True
the using .flat to set only the portion you need and do that with a list comprehension